python/Ganga/test/GPI/FileTests/TestLocalFileClient.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import absolute_import
+from Ganga.testlib.GangaUnitTest import GangaUnitTest
+from Ganga.testlib.file_utils import generate_unique_temp_file
+
+import datetime
+import time
+import os
+import shutil
+import copy
+
+class TestLocalFileClient(GangaUnitTest):
+    """test for sjid in filename names explain each test"""
+
+    _managed_files = []
+
+    # Num of sj in tests
+    sj_len = 3
+
+    # This sets up a LocalFileConfiguration which works by placing a file on local storage somewhere we can test using standard tools
+    LocalFileConfig = {'fileExtensions': [''],
+                       'uploadOptions': {},
+                       'backendPostprocess': {'LSF': 'client', 'LCG': 'client', 'ARC': 'client', 'Dirac': 'client',
+                                              'PBS': 'client', 'Interactive': 'client', 'Local': 'client', 'CREAM': 'client'}}
+
+    def setUp(self):
+        """
+        Configure the LocalFile for the test
+        """
+        extra_opts=[('PollThread', 'autostart', 'False'),
+                    ('Local', 'remove_workdir', 'False'),
+                    ('TestingFramework', 'AutoCleanup', 'False'),
+                    ('Output', 'LocalFile', self.LocalFileConfig),
+                    ('Output', 'FailJobIfNoOutputMatched', 'True')]
+        super(TestLocalFileClient, self).setUp(extra_opts=extra_opts)
+
+    @staticmethod
+    def cleanUp():
+        """ Cleanup the current temp objects """
+
+        from Ganga.GPI import jobs
+        for j in jobs:
+            shutil.rmtree(j.backend.workdir, ignore_errors=True)
+            j.remove()
+
+        for file_ in TestLocalFileClient._managed_files:
+            os.unlink(file_)
+        TestLocalFileClient._managed_files = []
+
+    def test_a_testClientSideSubmit(self):
+        """Test the client side code whilst stil using the Local backend"""
+
+        from Ganga.GPI import LocalFile, Job, ArgSplitter
+
+        TestLocalFileClient.cleanUp()
+
+        _ext = '.root'
+        file_1 = generate_unique_temp_file(_ext)
+        file_2 = generate_unique_temp_file(_ext)
+        TestLocalFileClient._managed_files.append(file_1)
+        TestLocalFileClient._managed_files.append(file_2)
+
+        j = Job()
+        j.inputfiles = [LocalFile(file_1), LocalFile(file_2)]
+        j.splitter = ArgSplitter(args = [[_] for _ in range(0, TestLocalFileClient.sj_len) ])
+        j.outputfiles = [LocalFile(namePattern='*'+_ext)]
+        j.submit()
+
+    def test_b_testClientSideComplete(self):
```

Comment:
In the old test system multipass tests were reserved for those tests which really *needed* to be multipass. All other tests (the majority) were .gpi files which were run independently. We still have some tests which need to by 'multipass' and for those the best solution is to name them `test_a_...`, `test_b_...` but they are in the minority.

It is prevalent in the Python testing world (and in all testing worlds really) that test functions should be runnable independently and in any order. This is why, for example, there's a `--last-failed` flag for pytest which only runs the tests that failed 'last time'. This breaks if we have dependent or ordered tests.

These tests here in this PR do not need to be multipass. The contents of `test_a_testClientSideSubmit` would be better described as a fixture (which could perhaps be parametrised and shared across test functions), `test_b_testClientSideComplete` would be a single test function, parametrised on the `LocalFileConfig['backendPostprocess']['Local']` setting and `cleanUp` would be a finaliser for the fixture. It would simplify the test code, make it more composable and extensible and allow the advertised test functions to be run independently in any order.